test/integration/rmi_out_test.py

```diff
@@ -0,0 +1,282 @@
+    ppl = rmi_out.grab_plant_part_list(clobber=clobber)
+    assert not ppl.empty
+    for ppl_df in ["plant_parts_eia", "gens_mega_eia", "true_grans_eia"]:
+        if ppl_df in rmi_out.pudl_out._dfs:
+            del rmi_out.pudl_out._dfs[ppl_df]
+
+
+def test_deprish_out(rmi_out, request):
+    """Test compilation of depreciation data."""
+    clobber = not request.config.getoption("--cached-deprish")
+    deprish = rmi_out.grab_deprish(clobber=clobber)
+    assert not deprish.empty
+
+
+def test_deprish_to_eia_out(rmi_out, request):
+    """Test fuzzy matching of depreciation data to EIA Plant Parts List."""
+    clobber = not request.config.getoption("--cached-deprish-eia")
+    deprish_to_eia = rmi_out.grab_deprish_to_eia(clobber=clobber)
+    assert not deprish_to_eia.empty
+
+
+def test_ferc1_to_eia(rmi_out, request):
+    """Test linkage of FERC 1 data to EIA PPL using record linkage."""
+    clobber = not request.config.getoption("--cached-ferc1-eia")
+    ferc1_to_eia = rmi_out.grab_ferc1_to_eia(clobber=clobber)
+    assert not ferc1_to_eia.empty
+
+
+def test_deprish_to_ferc1(rmi_out):
+    """Test linkage of Depriciation data to FERC 1 data."""
+    deprish_to_ferc1 = rmi_out.grab_deprish_to_ferc1(clobber=True)
+    assert not deprish_to_ferc1.empty
+
+
+##################
+# Validation Tests
+##################
+
+
+def gb_prep(
+    df1: pd.DataFrame, df2: pd.DataFrame, data_col: str, by: List[str]
+) -> pd.DataFrame:
+    """
+    Merge two grouped input tables to determine if summed data column matches.
+
+    TODO: Please help me name this....
+
+    Args:
+        df1: One dataframe to sum and check consistency with ``df2``.
+        df2: Other dataframe to sum and check consistency against ``df1``.
+        data_col: data column to check. Column must be in both ``df1`` and
+            ``df2``.
+    """
+    return (
+        pd.merge(
+            group_sum_col(df1, data_col=data_col, by=by),
+            group_sum_col(df2, data_col=data_col, by=by),
+            right_index=True,
+            left_index=True,
+            suffixes=("_1", "_2"),
+            how="outer",
+        )
+        .assign(
+            match=lambda x: np.isclose(
+                x[f"{data_col}_1"], x[f"{data_col}_2"], equal_nan=True
+            ),
+            off_rate=lambda x: x[f"{data_col}_1"] / x[f"{data_col}_2"],
+        )
+        .sort_index()
+    )
+
+
+def group_sum_col(df, data_col: str, by: List[str]) -> pd.DataFrame:
+    """Groupby sum a specific table's data col."""
+    gb_out = (
+        df.assign(  # convert date to year bc many of the og depish studies are EOY
+            report_year=lambda x: x.report_date.dt.year
+        )
+        .astype({"report_year": pd.Int64Dtype()})[
+            df.plant_id_eia.notnull()
+        ]  # only plant associated reocrds
+        .groupby(by=by, dropna=True)[[data_col]]
+        .sum(min_count=1)
+    )
+    return gb_out
```

Comment:
Using `gb_*` makes it seem like you're returning a `GroupBy` object, but it's getting aggregated, so it's a dataframe.